### PR TITLE
[VR-12836] upload_completed added in artifact entity

### PR DESCRIPTION
**Pull request by anandjakhaniya**
Jira Ticket: https://vertaai.atlassian.net/browse/VR-12836

### CODE REVIEW

**Review by conradoverta — COMMENTED** [review]
Don't we have to set this flag at some point?

**Review by conradoverta — COMMENTED** [review]
More specifically, only when committed. The issue is exactly that the artifact might not be complete when we just log it. Example:
1. log
2. try to push to aws and fail

**Review by conradoverta — CHANGES_REQUESTED** [review]
We don't seem to be setting `upload_completed` during the final commit call, right?

**Comment by anandjakhaniya** [comment]
> We don't seem to be setting `upload_completed` during the final commit call, right?

No, We should have to set it. In current implementation, we are setting  `upload_completed` during `commitMultipartArtifact` https://github.com/VertaAI/modeldb/blob/15017c59c907214e128a6d33f79f0285ef1a8320/backend/src/main/java/ai/verta/modeldb/experimentRun/ExperimentRunDAORdbImpl.java#L2647

and for new implementation at here https://github.com/VertaAI/modeldb/blob/15017c59c907214e128a6d33f79f0285ef1a8320/backend/src/main/java/ai/verta/modeldb/experimentRun/subtypes/ArtifactHandler.java#L297

**Review by conradoverta — APPROVED** [review]

**Review by conradoverta — CHANGES_REQUESTED** [review]
Every pre-existing artifact should have the flag set to `true` as it was already uploaded before. Only new artifacts can have `upload_completed=false`.

File: backend/src/main/resources/liquibase/db-changelog-2.0.xml [hunk 1]
```diff
@@ -979,4 +979,24 @@
...
+        </preConditions>
+        <update tableName="artifact">
+            <column name="upload_completed" valueBoolean="true" />
+            <where>field_type="artifacts" AND upload_completed = 0</where>
+        </update>
+        <rollback>
+            <update tableName="artifact">
+                <column name="upload_completed" valueBoolean="false" />
+                <where>field_type="artifacts" AND upload_completed = 1</where>
+            </update>
```
Comment (conradoverta) [inline backend/src/main/resources/liquibase/db-changelog-2.0.xml]: This rollback is not valid. This is an information losing change, so we can't rollback from it. Let's remove to make that clearer.

**Review by conradoverta — APPROVED** [review]
One change. LGTM otherwise

File: backend/src/main/resources/liquibase/db-changelog-2.0.xml (hunk 1 above)
Comment (anandjakhaniya) [reply backend/src/main/resources/liquibase/db-changelog-2.0.xml]: Ok, got it. removed rollback tag now.

**Pull request merged**